src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -318,89 +326,70 @@ public List<JobModel> enumerateJobs(
-    HashMap<String, List<String>> roleMap = new HashMap<>();
-    ArrayList<FlightState> flightStateList = new ArrayList<>();
-    int numTimesQueried = 0;
-    while (flightStateList.size() < offset + limit && numTimesQueried < MAX_FLIGHT_SEARCH_QUERIES) {
-      logger.info("Getting flight batch {} with nextPageToken {}", numTimesQueried, nextPageToken);
-      FlightEnumeration filterResults =
-          stairway.getFlights(nextPageToken, FLIGHT_SEARCH_BATCH_SIZE, filter);
-      if (filterResults.getFlightStateList().size() == 0) {
-        break;
-      }
-
-      for (var flightState : filterResults.getFlightStateList()) {
-        if (userLaunchedFlight(flightState, userReq)) {
-          flightStateList.add(flightState);
-        } else {
-          FlightMap inputParameters = flightState.getInputParameters();
-          IamResourceType resourceType =
-              inputParameters.get(JobMapKeys.IAM_RESOURCE_TYPE.getKeyName(), IamResourceType.class);
-          String resourceId =
-              inputParameters.get(JobMapKeys.IAM_RESOURCE_ID.getKeyName(), String.class);
-          IamAction action =
-              inputParameters.get(JobMapKeys.IAM_ACTION.getKeyName(), IamAction.class);
-          if (resourceType != null & resourceId != null && action != null) {
-            String key = String.format("%s-%s", resourceType, resourceId);
-            List<String> userRoles = roleMap.get(key);
-            if (userRoles == null) {
-              userRoles = samService.listActions(userReq, resourceType, resourceId);
-              roleMap.put(key, userRoles);
-            }
-            if (userRoles.contains(action.toString())) {
-              flightStateList.add(flightState);
-            }
-          }
-        }
-        // Stop processing as soon as we've got enough to return
-        if (flightStateList.size() == offset + limit) {
-          break;
-        }
-      }
-      numTimesQueried++;
-      nextPageToken = filterResults.getNextPageToken();
+  private FlightBooleanOperationExpression createFlightFilter(
+      AuthenticatedUserRequest userReq, String className) {
+    List<FlightFilterPredicateInterface> topLevelBooleans = new ArrayList<>();
+    // Exclude FileIngestWorkerFlight subflights since they pollute the jobs page.  We may
+    // eventually add a boolean to re-add or display the subflights as children.
+    // Note: do not add this filter if the user is explicitly requesting FileIngestWorkerFlight's
+    if (!StringUtils.isEmpty(className)) {
+      topLevelBooleans.add(makePredicateFlightClass(FlightFilterOp.EQUAL, className));
     }
-    if (flightStateList.size() <= offset) {
-      return new ArrayList<>();
-    } else if (flightStateList.size() < offset + limit) {
-      return flightStateList.subList(offset, flightStateList.size());
+    if (!StringUtils.isEmpty(className) && !className.equals(FileIngestWorkerFlight.class.getName())
+        || StringUtils.isEmpty(className)) {
```

Comment:
I think you can swap the order of `isEmpty()` here and avoid one of the calls
```suggestion
    if (StringUtils.isEmpty(className) || !className.equals(FileIngestWorkerFlight.class.getName()) {
```

Reply:
ah right